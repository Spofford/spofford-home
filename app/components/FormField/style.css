.inputGroup .error-message {
  color:red;
  font-family: 'Libre Baskerville', serif;
  font-size:.8em;
  text-transform: none;
}

.error-message {
  margin-top:2px;
  height:35px;
}

.signup button:disabled {
  background-color:white;
  box-shadow:none;
  color:#999;
}
