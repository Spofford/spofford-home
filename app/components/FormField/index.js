import React from "react";
import FontAwesome from "react-fontawesome";
import cssModules from 'react-css-modules';
import style from "./style.css";

export class FormField extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      errorMessage: ""
    }

    this.inlineValidation = this.inlineValidation.bind(this);
  }

  inlineValidation(event) {
    var element = document.getElementById(this.props.id)
    var responseObj = {name:event.target.name}
    if (event.target.value == "") {
      element.style = "border-color:red"
      responseObj.bool = false
      this.props.onValidate(responseObj)
    } else {
      if (event.target.name=="email") {
        var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        if (!re.test(event.target.value)) {
          element.style = "border-color:red"
          this.setState({errorMessage: "Not a valid email address"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="passwordConfirmation") {
        if (event.target.value != document.getElementById("signup-password").value) {
          element.style = "border-color:red"
          this.setState({errorMessage: "Passwords do not match"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="password") {
        if (event.target.value.length < 8) {
          element.style = "border-color:red"
          this.setState({errorMessage: "8 characters required"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="firstName") {
        element.style = "border-color:#ccc"
        this.setState({errorMessage: ""})
        responseObj.bool = true
        this.props.onValidate(responseObj)
      }
      if (event.target.name=="lastName") {
        element.style = "border-color:#ccc"
        this.setState({errorMessage: ""})
        responseObj.bool = true
        this.props.onValidate(responseObj)
      }
    }

  }


  render() {
    return (
      <div className="inputGroup">
        <label>{this.props.label}
          <input
            name={this.props.name}
            id={this.props.id}
            className="input"
            type={this.props.type}
            autoComplete={this.props.autocomplete}
            onChange={this.inlineValidation}
          />
          <div className="error-message">{this.state.errorMessage}</div>
        </label>
      </div>
    )
  }
};

export default cssModules(FormField, style);
