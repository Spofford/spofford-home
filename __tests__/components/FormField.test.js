import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import FormField from '../../app/components/FormField';

describe('<FormField />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<FormField />);
      //const component = wrapper.dive();

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    // test to render model from contentful
  });
});
