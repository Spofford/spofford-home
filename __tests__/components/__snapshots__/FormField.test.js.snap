// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> render() renders the component 1`] = `
<div
  className="inputGroup"
>
  <label>
    <input
      className="input"
      onChange={[Function]}
    />
    <div
      className="error-message"
    />
  </label>
</div>
`;
